File: packages/python/qsu/file/getFilePathLevel.py
import re
from .toPosixFilePath import toPosixFilePath


def getFilePathLevel(filePath: str) -> int:
	if not filePath:
		return -1

	if filePath == '/':
		return 1

	# Strip trailing separators of either flavour so that '/home/user' and
	# '/home/user/' report the same level.
	return len(toPosixFilePath(re.sub(r'[\\/]+$', '', filePath)).split('/'))

File: packages/javascript/lib/node/file/getFilePathLevel.ts
import { posix } from 'path';
import { toPosixFilePath } from './toPosixFilePath.js';

export function getFilePathLevel(filePath: string): number {
	if (!filePath) {
		return -1;
	}

	if (filePath === '/') {
		return 1;
	}

	// Strip trailing separators of either flavour so that '/home/user' and
	// '/home/user/' report the same level.
	return toPosixFilePath(filePath.replace(/[\\/]+$/, '')).split(posix.sep).length;
}
